Sources/OpenAPIRuntime/Conversion/CurrencyExtensions.swift

```diff
@@ -158,12 +158,56 @@ extension Converter {
     ) throws -> T {
         try decoder.decode(T.self, from: data)
     }
+    
+    func convertURLEncodedFormToCodable<T:Decodable>(
+        _ data: Data
+    ) throws -> T {
+        // convert data to string
+        guard let urlFormString = String(data: data, encoding: .utf8) else {
+            throw RuntimeError.failedToSerializeCodableData(type: String(describing: T.self))
+        }
+        
+        //separate string into dictionary by = and &
+        var dictionary: [String: String] = [:]
+        let pairs = urlFormString.split(separator: "&")
+        for pair in pairs {
+            let components = pair.split(separator: "=")
+            let key = String(components[0])
+            let value = String(components[1])
+            dictionary[key] = value
+        }
+        
+        //deserialize into JSON representation
+        let jsonData = try JSONSerialization.data(withJSONObject: dictionary)
```

Comment:
What happens if the Codable type has a property `foo: Int`? I suspect it won't parse, as in the JSON it'll be `"1"` instead of `1`? Same for other non-string primitive types that have a native JSON type?

Reply:
Yes, good spot - we certainly need some unit tests to cover non trivial conversions here, and this implementation has a fair few limitations!